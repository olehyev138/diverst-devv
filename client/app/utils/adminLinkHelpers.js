import { ROUTES } from 'containers/Shared/Routes/constants';

const NameToPathMap = Object.freeze({
  metrics_overview: ROUTES.admin.analyze.overview,
  groups_create: ROUTES.admin.manage.groups.index,
  segments_create: ROUTES.admin.manage.segments.index,
  groups_calendars: ROUTES.admin.manage.calendar.index,
  enterprise_folders_view: ROUTES.admin.manage.resources.index,
  groups_budget_manage: ROUTES.admin.plan.budgeting.index,
  campaigns_create: ROUTES.admin.innovate.campaigns.index,
  polls_create: ROUTES.admin.include.polls.index,
  mentoring_interests_manage: null,
  users_create: ROUTES.admin.system.users.index,
  sso_authentication: ROUTES.admin.system.globalSettings.ssoSettings.edit,
  policy_templates_manage: ROUTES.admin.system.users.policy_templates.index,
  fields_manage: ROUTES.admin.system.globalSettings.fields.index,
  branding_manage: ROUTES.admin.system.branding.index,
  custom_text_manage: ROUTES.admin.system.globalSettings.customText.edit,
  emails_manage: ROUTES.admin.system.globalSettings.emails.layouts.index,
  integrations_manage: null,
  rewards_manage: null,
  logs_view: ROUTES.admin.system.logs.index,
  edit_posts: null,
});

const GlobalSettingsPaths = Object.freeze([
  'users_manage',
  'sso_authentication',
  'policy_templates_manage',
  'fields_manage',
  'branding_manage',
  'custom_text_manage',
  'emails_manage',
  'integrations_manage',
  'rewards_manage',
  'logs_manage',
  'edit_posts',
]);

const GroupBudgetManagePaths = Object.freeze([
  'groups_budget_manage',
]);

const GroupManagePaths = Object.freeze([
  'groups_create',
  'segments_create',
  'groups_calendars',
  'enterprise_folders_view',
]);

const RootManagePaths = Object.freeze([
  ...GroupManagePaths,
  'metrics_overview',
  ...GroupBudgetManagePaths,
  'campaigns_create',
  'polls_create',
  'mentoring_interests_manage',
  ...GlobalSettingsPaths,
]);

export function resolveRootManagePath(permission) {
  const page = permission && RootManagePaths.find(page => permission[page] && NameToPathMap[page]);
  return page ? NameToPathMap[page].path() : null;
}
