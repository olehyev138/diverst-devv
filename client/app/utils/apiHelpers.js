export function appendQueryArgs(url, opts) {
  let newUrl = url;

  // append query arguments
  if (opts) {
    newUrl += '?';
    for (const arg of Object.keys(opts)) {
      if (newUrl.indexOf('?') !== newUrl.length - 1)
        newUrl += '&';

      if (Array.isArray(opts[arg]))
        newUrl += `${arg}=${JSON.stringify(opts[arg])}`;
      else
        newUrl += `${arg}=${opts[arg]}`;
    }
  }

  return newUrl;
}
