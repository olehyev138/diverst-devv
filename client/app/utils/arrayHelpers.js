export const insertIntoArray = (arr, value) => arr.reduce((result, element, index, array) => {
  result.push(element);
  if (index < array.length - 1)
    result.push(value);
  return result;
}, []);
