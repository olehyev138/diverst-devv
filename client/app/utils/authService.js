import { store } from 'configureStore';

const AuthService = {
  isAuthenticated() {
    const state = store.getState();

    if (state.global.token) return true;

    return false;
  },

  hasPermission(routeData) {
    if (!routeData)
      return true;

    return Object.prototype.hasOwnProperty.call(routeData, 'permission') === false || AuthService.getPolicyGroup()[routeData.permission] === true;
  },

  getPolicyGroup() {
    const state = store.getState();

    return state.global.policy_group || undefined;
  },

  getEnterprise() {
    const state = store.getState();
    return state.global.enterprise || undefined;
  },
};

export default AuthService;
