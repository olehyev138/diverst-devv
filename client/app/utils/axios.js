const axios = require("axios");

export default function configureAxios() {
    // Add a request interceptor
    axios.interceptors.request.use(
        function(config) {
            // Do something before request is sent
            return config;
        },
        function(error) {
            // Do something with request error
            return Promise.reject(error);
        },
    );

    // Add a response interceptor
    axios.interceptors.response.use(
        function(response) {
            // Do something with response data
            return response;
        },
        function(rejection) {
            if (!rejection.response) {
                rejection.response = {
                    data: "Server Unavailable"
                };
            }
            else if (rejection.response.data && rejection.response.data.message) {
                rejection.response.data = rejection.response.data.message;
            }
            else if (rejection.response.status == 500) {
                rejection.response.data =
                    "Internal Server Error. Please contact support@diverst.com";
            }
            else if (rejection.response.status == 503) {
                rejection.response.data =
                    "Server Unavailable";
            }

            // Do something with response error
            return Promise.reject(rejection);
        },
    );
}
