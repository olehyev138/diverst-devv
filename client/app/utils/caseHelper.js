export default (object, ...args) => {
  const valid = args.find(condition => equals(condition[0], object) || condition[0] == null);
  if (valid)
    return evaluate(valid[1], object);
  return null;
};

const evaluate = (predicate, object = null) => {
  if (typeof predicate === 'function')
    return predicate(object);
  return predicate;
};

const equals = (predicate, object) => {
  if (predicate instanceof Array)
    return predicate.some(sub => equals(sub, object));
  if (typeof predicate === 'function')
    return predicate(object);
  return predicate === object;
};
