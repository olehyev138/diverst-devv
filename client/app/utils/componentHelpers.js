import React from 'react';

export const renderChildrenWithProps = (children, props) => React.Children.map(children, child => React.cloneElement(child, { ...props }));
