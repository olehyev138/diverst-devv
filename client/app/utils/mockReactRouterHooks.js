// This file is intended to be used for when a component uses react-router hooks that require the router context,
// such as `useLocation`, but you're using `shallow` to test since you only care about how the component itself renders,
// this will stop those errors from happening while still allowing you to use `shallow` and have the snapshot work properly.
//
// Important: Must be imported before the component being tested is imported.

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/',
    key: 'testKey',
  })
}));
