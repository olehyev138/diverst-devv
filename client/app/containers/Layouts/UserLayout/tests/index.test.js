import React from 'react';
import { shallow } from 'enzyme';
import { unwrap } from '@material-ui/core/test-utils';
import 'utils/mockReactRouterHooks';

import { StyledUserLayout } from '../index';
const UserLayoutNaked = unwrap(StyledUserLayout);

/**
 * TODO:
 *  - test it accepts & renders a child component
 *  - test it renders AuthenticatedLayout as a child component
 *  - test it renders UserLinks as a child component
 */


const props = {
  classes: {},
  component: jest.fn(),
  pageTitle: {}
};

describe('<UserLayout />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = shallow(<UserLayoutNaked {...props} />);

    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const wrapper = shallow(<UserLayoutNaked {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
