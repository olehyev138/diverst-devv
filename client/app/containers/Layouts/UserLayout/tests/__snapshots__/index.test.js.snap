// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserLayout /> Should render and match the snapshot 1`] = `
<WithStyles(AuthenticatedLayout)
  component={[Function]}
  pageTitle={Object {}}
  position="absolute"
/>
`;
